### PR TITLE
feat(threat-model): create LinkScriptProvider to add event listener on target page links

### PR DESCRIPTION
**Pull request by lisli1**
#### Description of changes

Created LinkScriptProvider that adds an event listener to the "click" event on elements with the `target-page-link` id in the generated HTML report. The event handler prompts the browser to generate a confirmation dialog, where selecting OK opens the target page in a new tab, and selecting Cancel prevents the target page from being opened.

Added relevant unit tests. There is no unit test for `getDefaultAddListenerForLink` because the snapshot text representation is a bit confusing (see `collapsible-script-provider.tsx` for similar issue).

Note: To keep this PR small, I will add the event listeners to the target page links in future PRs. As a result, there are currently no changes to UI.

#### Pull request checklist

- [x] Partially addresses an existing issue: # 1521197
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). Check workflow guide at: `<rootDir>/docs/workflow.md`
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/reports/components/report-sections/link-script-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+                'Cancel to stay on the current page.',
+        );
+
+        if (result === false) {
+            event.preventDefault();
+        }
+    });
+};
+
+export const getAddListenerForLink = (code: string | Function): string =>
```
Comment (smoralesd) [inline src/reports/components/report-sections/link-script-provider.tsx]: Besides the name, this is the same function we already have on 'src\reports\components\report-sections\collapsible-script-provider.tsx'. Should be good to extract this to it's own file and reuse on both places.

File: src/tests/unit/tests/reports/components/report-sections/link-script-provider.test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    getAddListenerForLink,
+} from 'reports/components/report-sections/link-script-provider';
+import { It, Mock, MockBehavior, Times } from 'typemoq';
+
+describe('LinkScriptProvider', () => {
+    describe('getAddListenerForLink', () => {
+        it('matches content', () => {
+            const code = 'this is test code';
+
+            const result = getAddListenerForLink(code);
```
Comment (smoralesd) [inline src/tests/unit/tests/reports/components/report-sections/link-script-provider.test.ts]: If we reuse the function from 'src\reports\components\report-sections\collapsible-script-provider.tsx', this test should be on a different place.

File: src/tests/unit/tests/reports/components/report-sections/link-script-provider.test.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            const targetPageLinkMock = Mock.ofType<HTMLElement>();
+            const docMock = Mock.ofType<Document>(undefined, MockBehavior.Strict);
+            docMock
+                .setup(doc => doc.getElementById('target-page-link'))
+                .returns(() => targetPageLinkMock.object);
+
+            addEventListenerForLink(docMock.object);
+
+            targetPageLinkMock.verify(
+                link => link.addEventListener('click', It.is(isFunction)),
```
Comment (smoralesd) [inline src/tests/unit/tests/reports/components/report-sections/link-script-provider.test.ts]: This `verify` probes we are adding and event listener to the proper element on the dom.
We are missing a test when we actually click that element, triggering the listener function. 
Doing so will likely make you change how we are using `confirm` as mention on the comment I left on the actual usage of `confirm`

File: src/reports/components/report-sections/link-script-provider.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+export const addEventListenerForLink = (doc: Document) => {
+    const targetPageLink = doc.getElementById('target-page-link');
+
+    targetPageLink.addEventListener('click', event => {
+        const result = confirm(
```
Comment (smoralesd) [inline src/reports/components/report-sections/link-script-provider.tsx]: Here, `confirm` is being use as a global dependency which is not ideal. One way to solve this is to do the same you are doing with the `doc: Document` param. That way, you can mock the function on the unit tests.

File: src/reports/components/report-sections/link-script-provider.tsx (hunk 1 above)
Comment (smoralesd) [reply src/reports/components/report-sections/link-script-provider.tsx]: On a second though, we may need a slightly different function to consider how we pass the reference to the `confirm` function.

File: src/tests/unit/tests/reports/components/report-sections/link-script-provider.test.ts (hunk 2 above)
Comment (lisli1) [reply src/tests/unit/tests/reports/components/report-sections/link-script-provider.test.ts]: Changed the function slightly, will be leaving the tests separate

File: src/reports/components/report-sections/link-script-provider.tsx (hunk 1 above)
Comment (lisli1) [reply src/reports/components/report-sections/link-script-provider.tsx]: Changed the function slightly to include a reference to `confirm`

**Pull request closed without merging**